waku/waku_archive/archive.nim

```diff
@@ -163,7 +166,9 @@ proc findMessages*(
   let pageSize = min(rows.len, int(maxPageSize))
 
   #TODO once store v2 is removed, unzip instead of 2x map
-  messages = rows[0 ..< pageSize].mapIt(it[1])
+  if query.includeData:
```

Comment:
This perhaps can be done in a separate PR, but a potential improvement for this code could be to consider modifying the `getMessages` function in line 139 to also pass the `includeData`, and then, in the drivers define a query that will only do a  `SELECT messageHash FROM messages .......` if the value from `includeData` is `true`.

We should check the query plan from sqlite/postgres, of course, but depending on how the indexes were defined, this should be more efficient, due to reduced disk i/o and data transfer from disk to memory, since the query will be satisfied by only reading the values from the index, without needing to fetch additional data from the table. I imagine this would be the case as the following indexes are defined in the code as far as i could see:
- Primary Key: messageHash
- Index: contentTopic, pubsubTopic, senderTimestamp, id
- Index: storedAt

Reply:
I'll create an issue to not lose track of this then!

Reply:
In retrospect, I try to share as much code with v2 as possible an it was a mistake.